rust/libnewsboat-ffi/src/filepath.rs

```diff
@@ -101,19 +102,15 @@ fn starts_with(filepath: &PathBuf, str: &str) -> bool {
     filepath.0.starts_with(str)
 }
 
-fn file_name(filepath: &PathBuf) -> Vec<u8> {
-    use std::str::FromStr;
+fn file_name(filepath: &PathBuf) -> Box<PathBuf> {
     if filepath.0.file_name().is_some() {
-        filepath
-            .0
-            .file_name()
-            .unwrap()
-            .to_os_string()
-            .into_string()
-            .unwrap()
-            .as_bytes()
-            .to_vec()
+        let res = filepath.0.file_name().unwrap();
```

Comment:
Please use `if let Some(res) = filepath.0.file_name() { ... }` — this avoids the call to `unwrap()`.